opnode/rollup/driver/step.go

```diff
@@ -212,59 +213,51 @@ func (d *outputImpl) insertEpoch(ctx context.Context, l2Head eth.L2BlockRef, l2S
 
 // attributesMatchBlock checks if the L2 attributes pre-inputs match the output
 // nil if it is a match. If err is not nil, the error contains the reason for the mismatch
-func attributesMatchBlock(attrs *l2.PayloadAttributes, parentHash common.Hash, block *types.Block) error {
-	if parentHash != block.ParentHash() {
-		return fmt.Errorf("parent hash field does not match. expected: %v. got: %v", parentHash, block.ParentHash())
+func attributesMatchBlock(attrs *l2.PayloadAttributes, parentHash common.Hash, block *l2.ExecutionPayload) error {
+	if parentHash != block.ParentHash {
+		return fmt.Errorf("parent hash field does not match. expected: %v. got: %v", parentHash, block.ParentHash)
 	}
-	if uint64(attrs.Timestamp) != block.Time() {
-		return fmt.Errorf("timestamp field does not match. expected: %v. got: %v", uint64(attrs.Timestamp), block.Time())
+	if attrs.Timestamp != block.Timestamp {
+		return fmt.Errorf("timestamp field does not match. expected: %v. got: %v", uint64(attrs.Timestamp), block.Timestamp)
 	}
-	if attrs.PrevRandao != l2.Bytes32(block.MixDigest()) {
-		return fmt.Errorf("random field does not match. expected: %v. got: %v", attrs.PrevRandao, l2.Bytes32(block.MixDigest()))
+	if attrs.PrevRandao != block.PrevRandao {
+		return fmt.Errorf("random field does not match. expected: %v. got: %v", attrs.PrevRandao, block.PrevRandao)
 	}
-	if len(attrs.Transactions) != len(block.Transactions()) {
-		return fmt.Errorf("transaction count does not match. expected: %v. got: %v", len(attrs.Transactions), len(block.Transactions()))
+	if len(attrs.Transactions) != len(block.Transactions) {
+		return fmt.Errorf("transaction count does not match. expected: %v. got: %v", len(attrs.Transactions), block.Transactions)
 	}
-	btxs := block.Transactions()
-	for i := range attrs.Transactions {
-		var tx types.Transaction
-		err := tx.UnmarshalBinary(attrs.Transactions[i])
-		if err != nil {
-			return fmt.Errorf("failed to decode transaction %d in attributes: %w", i, err)
-		}
-
-		if tx.Hash() != btxs[i].Hash() {
-			return fmt.Errorf("transaction %d does not match. expected: %v. got: %v", i, tx.Hash(), btxs[i].Hash())
+	for i, otx := range attrs.Transactions {
```

Comment:
Nice here. The previous version was also hurt by the fact that the L2 EE was returning the wrong values